checks/kubernetes/resources/use_resource_quota.rego

```diff
@@ -11,27 +11,33 @@
 #   avd_id: AVD-KSV-0040
 #   severity: LOW
 #   short_code: resource-quota-usage
-#   recommended_action: "create resource quota policy with mem and cpu quota per each namespace"
+#   recommended_action: Create a ResourceQuota policy with memory and CPU quotas for each namespace
 #   input:
 #     selector:
 #     - type: kubernetes
+#       subtypes:
+#         - kind: resourcequota
 package builtin.kubernetes.KSV040
 
 import rego.v1
 
 import data.lib.kubernetes
 
-resourceQuotaConfigure if {
-	lower(input.kind) == "resourcequota"
-	input.spec[hard]
-	kubernetes.has_field(input.spec.hard, "requests.cpu")
-	kubernetes.has_field(input.spec.hard, "requests.memory")
-	kubernetes.has_field(input.spec.hard, "limits.cpu")
-	kubernetes.has_field(input.spec.hard, "limits.memory")
+required_fields := {
+	"requests.cpu",
+	"requests.memory",
+	"limits.cpu",
+	"limits.memory",
+}
+
+resource_quota_configured if {
+	every field in required_fields {
+		kubernetes.has_field(input.spec.hard, field)
+	}
```

Comment:
nice TIL! There might be other opportunities for us to refactor in a similar fashion. Certainly can be done in another PR.